cordova-lib/spec-cordova/platform.spec.js

```diff
@@ -356,3 +356,123 @@ describe('plugin add and rm end-to-end --fetch', function () {
+        shell.rm('-rf', tmpDir);
+    });
+
+    it('installs and uninstalls plugin depending on new pod and existing pod', function(done) {
+
+        cordova.raw.create('hello4')
+        .then(function() {
+            process.chdir(project);
+            //TODO: change this to cordova-ios on npm 
+            return cordova.raw.platform('add', 'https://github.com/juliascript/cordova-ios.git#CB-9825');
+        })
+        .then(function() {
+            return cordova.raw.plugin('add', samplePlugin);
+        })
+        .then(function() {
+            podfile = path.resolve('./platforms/ios/Podfile');
+            podsJSON = path.resolve('./platforms/ios/pods.json');
+            workspace = path.resolve('./platforms/ios/HelloCordova.xcworkspace');
+
+            //podfile should have been created
+            fs.exists(podfile, function(podfileExists){
+                expect(podfileExists);
+            });
+
+            //pods.json should have been created
+            fs.exists(podsJSON, function(podsJSONExists){
+                expect(podsJSONExists);
+            });
+
+            //workspace should have been created
+            fs.exists(workspace, function(workspaceCreated){
+                expect(workspaceCreated);
+            });
+
+            delete require.cache[require.resolve(podfile)];
+            var podfileContent = fs.readFileSync(podfile, {'encoding' : 'utf8'});
+
+            expect(podfileContent.includes(AFNetworking));
+
+            delete require.cache[require.resolve(podsJSON)];
+            var podsJSONContent = require(podsJSON);
+
+            expect(podsJSONContent[AFNetworking] !== null);
+            return cordova.raw.plugin('add', overlappingDependencyPlugin);
+        })
+        .then(function() {
+            delete require.cache[require.resolve(podfile)];
+            var podfileContent = fs.readFileSync(podfile, {'encoding' : 'utf8'}); 
+            var numberOfTimesAFNetworkingIsInPodfile = podfileContent.match(/AFNetworking/g || []).length;
+
+            expect(podfileContent.includes(CWStatusBarNotification));
+            expect(numberOfTimesAFNetworkingIsInPodfile).toEqual(1); 
+
+            delete require.cache[require.resolve(podsJSON)];
+            var podsJSONContent = require(podsJSON);
+            var countPropertyOfAFNetworkingInPodsJSON = podsJSONContent[AFNetworking].count;
+            var specPropertyOfAFNetworkingInPodsJSON = podsJSONContent[AFNetworking].spec;
+
+            expect(countPropertyOfAFNetworkingInPodsJSON).toEqual(2);
+            //spec property should not be changed because of overlapping dependency
+            expect(specPropertyOfAFNetworkingInPodsJSON).toEqual('~> 3.0');
+
+            return cordova.raw.plugin('rm','sample-cocoapod-plugin-overlapping-dependency');
+        })
+        .then(function() {
+            //expect only AFNetworking
+            delete require.cache[require.resolve(podfile)];
```

Comment:
The Podfile and pods.json should change after the removal and adding of a plugin. I believe the deletion of cache is necessary here.